pkgs/by-name/wo/wordlists/package.nix

```diff
@@ -1,45 +1,61 @@
 { lib
-, callPackage
+, symlinkJoin
 , nmap
 , rockyou
-, runtimeShell
 , seclists
-, symlinkJoin
-, tree
 , wfuzz
 , lists ? [
     nmap
     rockyou
     seclists
     wfuzz
   ]
+, writeShellScriptBin
+, tree
 }:
+let
+  version = "unstable-2023-11-14";
 
-symlinkJoin rec {
-  pname = "wordlists";
-  version = "unstable-2023-10-10";
-
-  name = "${pname}-${version}";
-  paths = lists;
+  wordlistsCollection = symlinkJoin {
+    name = "wordlists-collection-${version}";
+    paths = lists;
 
-  postBuild = ''
-    mkdir -p $out/bin
+    postBuild = let
+      pathsToKeep = lib.pipe [
+        "$out/share/wordlists/*"
+        "$out/share/wordlists"
+        "$out/share"
+        "$out"
+      ] [
+        (map (p: ''-path "${p}"''))
+        (builtins.concatStringsSep " -o ")
+      ];
+    in ''
+      find $out -type d ! \( ${pathsToKeep} \) -printf "rm -r \"%p\" 2>/dev/null || true\n" | sh
+    '';
+  };
 
-    # Create a command to show the location of the links.
-    cat >> $out/bin/wordlists << __EOF__
-    #!${runtimeShell}
-    ${tree}/bin/tree ${placeholder "out"}/share/wordlists
-    __EOF__
-    chmod +x $out/bin/wordlists
+  wordlistsBinaries = let
+    wordlistsBin = writeShellScriptBin "wordlists" ''
+      ${lib.getExe tree} ${wordlistsCollection}/share/wordlists
+    '';
+    wordlistsPathBin = writeShellScriptBin "wordlists_path" ''
+      printf "${wordlistsCollection}/share/wordlists\n"
+    '';
```

Comment:
would you mind adding the comments back?

Reply:
Yes, completely forgot! Thanks